### PR TITLE
chore(deps): migrate electron-packager to @electron/packager 18.3.4

### LINKED ISSUE #30124: Replace deprecated `electron-packager`
### What would you like?

Replace the deprecated npm module [electron-packager](https://www.npmjs.com/package/electron-packager) with the renamed module [@electron/packager](https://www.npmjs.com/package/@electron/packager).

### Why is this needed?

The latest version of [electron-packager](https://www.npmjs.com/package/electron-packager) is [17.1.2](https://github.com/electron/packager/releases/tag/v17.1.2), [@electron/packager](https://www.npmjs.com/package/@electron/packager) covers >= [18.0.0](https://github.com/electron/packager/releases/tag/v18.0.0) so the new npm module name must be used when upgrading to `18.x`.

### Other

- Renovate deprecation list https://github.com/cypress-io/cypress/issues/3777

| Item                      | Info                                                                                            |
| ------------------------- | ------------------------------------------------------------------------------------------------ |
| Deprecated npm module     | [electron-packager](https://www.npmjs.com/package/electron-packager)                             |
| npm registry message      | Please use @electron/packager moving forward. There is no API change, just a package name change |
| GitHub repo               | https://github.com/electron/packager                                                             |
| Repo status               | active                                                                                           |
| Replacement npm module    | https://www.npmjs.com/package/@electron/packager                                                 |
| Replacement repo          | no change, as above                                                                              |
| Replacement documentation | https://electron.github.io/packager/main/repo                                                    |
| Prerequisites             | Node.js `v16.13.0` (met), electron-packager >= `18.0.0`                                          |

Where used:

| Directory                                                                                 | Version                       |
| ----------------------------------------------------------------------------------------- | ----------------------------- |
| [packages/electron](https://github.com/cypress-io/cypress/tree/develop/packages/electron) | "electron-packager": "17.1.2" |

### PR DESCRIPTION
**Pull request by MikeMcC399**
- Closes  https://github.com/cypress-io/cypress/issues/30124

### Additional details

Replace the deprecated npm module [electron-packager](https://www.npmjs.com/package/electron-packager) `17.1.2` in [packages/electron](https://github.com/cypress-io/cypress/tree/develop/packages/electron) with the renamed module [@electron/packager](https://www.npmjs.com/package/@electron/packager) [18.3.4](https://github.com/electron/packager/releases/tag/v18.3.4).

The npm module [electron-packager](https://www.npmjs.com/package/electron-packager) is deprecated and shows the message:

> Please use @electron/packager moving forward. There is no API change, just a package name change

### Steps to test

```shell
git clean -xfd
yarn
yarn workspace @packages/electron build
yarn workspace @packages/electron test
```

### How has the user experience changed?

No change to user experience.

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/electron/lib/install.js [hunk 1]
```diff
@@ -138,10 +138,12 @@ module.exports = {
      * our mksnapshot. Requiring the package in this way, dynamically, will
      * make it undiscoverable by mksnapshot, which is OK since electron-packager
      * is a build dependency.
+     * Converted to use @electron/packaer for >= v18.x.x.
```
Comment (jennifer-shehane) [inline packages/electron/lib/install.js]:
```suggestion
     * Converted to use @electron/packager for >= v18.x.x.
```

**Comment by jennifer-shehane** [comment]
Running tests

**Comment by MikeMcC399** [comment]
All tests have passed
image

Updating branch again.

**Comment by jennifer-shehane** [comment]
@MikeMcC399 Hmm, that's not a typical error I see. Unsure if it's related.

Screenshot 2024-09-04 at 1 12 11 PM

**Comment by MikeMcC399** [comment]
@jennifer-shehane 
> Hmm, that's not a typical error I see. Unsure if it's related.

I suggest to run the tests again to see if the error is reproducible.

**Comment by MikeMcC399** [comment]
I successfully ran

```shell
yarn binary-build
yarn binary-package
yarn binary-zip
```

according to https://github.com/cypress-io/cypress/blob/develop/guides/building-release-artifacts.md

**Comment by jennifer-shehane** [comment]
@MikeMcC399 Running tests

**Review by jennifer-shehane — APPROVED** [review]
Noting breaking changes in 18: https://github.com/electron/packager/releases/tag/v18.0.0

**Comment by MikeMcC399** [comment]
CircleCI had an incident which was supposed to be resolved on Sep 11, 11:30 UTC
Attempting re-run.

**Comment by jennifer-shehane** [comment]
@MikeMcC399 Could you please pause on pulling in develop?

**Comment by MikeMcC399** [comment]
@jennifer-shehane 

> Could you please pause on pulling in develop?

Paused.

Should I leave this completely to you now? Just let me know if / when I should do anything else.

**Comment by MikeMcC399** [comment]
@jennifer-shehane 

A new version was released on Sep 11, 2024:

https://github.com/electron/packager/releases/tag/v18.3.5

Do you want to update to this one whilst the PR is still open?

**Comment by jennifer-shehane** [comment]
Let just get this in since it's been open for so long and now rdy

**Comment by MikeMcC399** [comment]
@jennifer-shehane 

> Let just get this in since it's been open for so long and now rdy

That's fine. I just wanted to check with you. Renovate can probably take care of bumping to the latest if it's needed at any time.

**Review by ryanthemanuel — APPROVED** [review]

**Review by AtofStryker — APPROVED** [review]

**Pull request merged**